contracts/extensions/bridge-operator-governance/BridgeAdminOperator.sol

```diff
@@ -0,0 +1,418 @@
+      unchecked {
+        ++i;
+      }
+    }
+  }
+
+  /**
+   * @inheritdoc IQuorum
+   */
+  function minimumVoteWeight() public view virtual returns (uint256) {
+    return (_num * _totalWeight + _denom - 1) / _denom;
+  }
+
+  function _addBridgeOperators(
+    uint256[] memory voteWeights,
+    address[] memory governors,
+    address[] memory brigdeOperators
+  ) internal returns (bool[] memory addeds) {
+    uint256 length = brigdeOperators.length;
+    addeds = new bool[](length);
+
+    EnumerableSet.AddressSet storage governorSet = _getGovernorsSet();
+    mapping(address => address) storage _governorOf = _getGovernorOf();
+    EnumerableSet.AddressSet storage bridgeOperatorSet = _getBridgeOperatorSet();
+    mapping(address => BridgeOperatorInfo) storage governorToBridgeOperatorInfo = _getGovernorToBridgeOperatorInfo();
+
+    address governor;
+    uint96 voteWeight;
+    address bridgeOperator;
+    uint256 accumulateWeight;
+    BridgeOperatorInfo memory bridgeOperatorInfo;
+    for (uint256 i; i < length; ) {
+      governor = governors[i];
+      bridgeOperator = brigdeOperators[i];
+      voteWeight = voteWeights[i].toUint96();
+
+      _checkNonZeroAddress(governor);
+      _checkNonZeroAddress(bridgeOperator);
+      _sanityCheck(voteWeight, governor, bridgeOperator);
+
+      addeds[i] = bridgeOperatorSet.add(bridgeOperator);
+
+      if (addeds[i]) {
+        governorSet.add(governor);
+
+        accumulateWeight += voteWeight;
+
+        _governorOf[bridgeOperator] = governor;
+
+        bridgeOperatorInfo.addr = bridgeOperator;
+        bridgeOperatorInfo.voteWeight = voteWeight;
+        governorToBridgeOperatorInfo[governor] = bridgeOperatorInfo;
+      }
+
+      unchecked {
+        ++i;
+      }
+    }
+
+    _totalWeight += accumulateWeight;
+
+    emit BridgeOperatorSetModified(msg.sender, BridgeAction.Add);
+  }
+
+  function _removeBridgeOperators(
+    address[] memory bridgeOperators
+  ) internal nonDuplicate(bridgeOperators) returns (bool[] memory removeds) {
+    uint256 length = bridgeOperators.length;
+    removeds = new bool[](length);
+
+    mapping(address => address) storage _governorOf = _getGovernorOf();
+    EnumerableSet.AddressSet storage governorSet = _getGovernorsSet();
+    EnumerableSet.AddressSet storage _bridgeOperatorSet = _getBridgeOperatorSet();
+    mapping(address => BridgeOperatorInfo) storage governorToBridgeOperatorInfo = _getGovernorToBridgeOperatorInfo();
+
+    address governor;
+    address bridgeOperator;
+    uint256 accumulateWeight;
+    BridgeOperatorInfo memory bridgeOperatorInfo;
+    for (uint256 i; i < length; ) {
+      bridgeOperator = bridgeOperators[i];
+      governor = _governorOf[bridgeOperator];
+
+      _checkNonZeroAddress(governor);
+      _checkNonZeroAddress(bridgeOperator);
+
+      bridgeOperatorInfo = governorToBridgeOperatorInfo[governor];
+      assert(bridgeOperatorInfo.addr == bridgeOperator);
```

Comment:
I think `if... revert(CustomError)` is better.

Reply:
reused logic should be in internal function to avoid deployment cost.